lib/compress.js

```diff
@@ -987,12 +1015,6 @@ merge(Compressor.prototype, {
         def(AST_Conditional, function(compressor){
             return this.consequent.is_string(compressor) && this.alternative.is_string(compressor);
         });
-        def(AST_Call, function(compressor){
-            return compressor.option("unsafe")
-                && this.expression instanceof AST_SymbolRef
-                && this.expression.name == "String"
-                && this.expression.undeclared();
-        });
```

Comment:
This is unreachable, as any `String(...)` [would have been transformed](https://github.com/mishoo/UglifyJS2/blob/master/lib/compress.js#L2436-L2445) already if `unsafe` is enabled.